src/jobprogress.ts

```diff
@@ -69,10 +69,29 @@ export class JobProgress {
 
   private onJobRemoved(jobRemoved: Protocol.JobRemoved) {
       if (!this.isJob(jobRemoved.handle)) {
-        return;
+          return;
       }
       this.clearTimeout();
-      this.resolve(this.job);
+      if (!StatusSeverity.isOk(jobRemoved.status)) {
+          vscode.window.showErrorMessage(this.getErrorMessage(jobRemoved.status));
+          this.resolve();
```

Comment:
You already have a clause like this for error handling, so why not just reject with the error message?

Reply:
Good catch. Correcting.

Reply:
And now you need to adjust the tests :)

Reply:
@jrichter1 they are, build is green :)